proto/src/determined/api/v1/experiment.proto

```diff
@@ -297,6 +307,9 @@ message MetricBatchesResponse {
 
 // Request metrics from all trials at a progress point of progress.
 message TrialsSnapshotRequest {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "experiment_id", "batches_processed", "metric_name", "metric_type" ] }
```

Comment:
@hamidzr In the generated Swagger UI locally, it only seems to be marking experiment_id as a required field, and not the others. I can't see what I've done wrong here - I think it was you that suggested this change - do you perhaps see anything?

Reply:
could you share the generated swagger spec? Not sure why it would take one and not the others. When we generate the spec we want it to advertise these fields as camel case I saw something today that might suggest it's not applying the camelCase to these annotations as well. That might have a hand?

Reply:
I think experiment_id is marked as required implicitly because it is a path parameter. Compile without these annotations and it is still required. (according to `cat build/swagger/determined/api/v1/api.swagger.json | jq '.paths."/api/v1/experiments/{experimentId}/metrics-stream/trials-snapshot"'`). Peeking at create experiment looks like it's not generating quite what is expected either.

Reply:
this should address that https://github.com/determined-ai/determined/pull/1681